src/main/java/com/hubspot/jinjava/lib/tag/eager/EagerDoTag.java

```diff
@@ -0,0 +1,74 @@
+package com.hubspot.jinjava.lib.tag.eager;
+
+import com.hubspot.jinjava.interpret.JinjavaInterpreter;
+import com.hubspot.jinjava.interpret.TemplateSyntaxException;
+import com.hubspot.jinjava.lib.tag.DoTag;
+import com.hubspot.jinjava.tree.parse.TagToken;
+import com.hubspot.jinjava.util.ChunkResolver;
+import java.util.StringJoiner;
+import org.apache.commons.lang3.StringUtils;
+
+public class EagerDoTag extends EagerStateChangingTag<DoTag> {
+
+  public EagerDoTag() {
+    super(new DoTag());
+  }
+
+  public EagerDoTag(DoTag doTag) {
+    super(doTag);
+  }
+
+  @Override
+  public String getEagerTagImage(TagToken tagToken, JinjavaInterpreter interpreter) {
+    String expr = tagToken.getHelpers();
+    if (StringUtils.isBlank(expr)) {
+      throw new TemplateSyntaxException(
+        interpreter,
+        tagToken.getImage(),
+        "Tag 'print' expects expression"
+      );
+    }
+    ChunkResolver chunkResolver = new ChunkResolver(expr, tagToken, interpreter);
+    EagerStringResult resolvedExpression = executeInChildContext(
+      eagerInterpreter -> chunkResolver.resolveChunks(),
+      interpreter,
+      true
+    );
+    StringJoiner joiner = new StringJoiner(" ");
```

Comment:
Looks like this logic is the same for both tags, and maybe others? Should we split it out into a funciton?

Reply:
Yes, that is a good call.